meetings/working-groups/collection-literals/Core-interface-target-type-proposal.md

```diff
@@ -0,0 +1,247 @@
+
+    It would also be highly unfortunate if we shipped and our own analyzers immediately flagged the usage as being a problem.
+
+    Finally, part of performance means being a good .Net citizen.  So the collections we produce should be able to pick up the optimizations the BCL has today for collection types.
+
+5. Safety.  Specifically, keeping data safe from undesirable mutation. We broadly think of users as being in two categories.  The first category generally doesn't think about safety, and thus would suffice with any solution on our part.  If mutable data is exposed or not isn't relevant to them as they do not care about keeping their data safe.  However, the second group (which is likely smaller, but present, vocal, and influential) absolutely wants to keep their exposed data safe from mutation.  Similar to the perf concerns, we believe we need a solution for this group that fits their expectations, puts them at ease, and isn't immediately banned (or flagged by analyzers) for doing unsafe things.
+
+## Options
+
+Note: there are truly a plethora of options available here.  However, based on feedback from LDM and the working group meetings, we tried to whittle things down to a reasonable few that we feel warrant discussion and comparison.
+
+### Option 1: Disallow target typing to these interface types.
+
+This is the simplest and cheapest option we have at our disposal.  We could just say that this is an error, and force the user to specify the type they want to generate.  However, we believe this produces a negative result for every factor at play above.
+
+Specifically:
+
+```c#
+void DoSomething(IEnumerable<int> values) { ... }
+
+// Not allowed.
+DoSomething([1, 2, 3]);
+
+// Write this instead:
+DoSomething((List<int>)[1, 2, 3]);
+```
+
+and
+
+```c#
+class Order
+{
+    // Not allowed:
+    public IReadOnlyList<OrderTag> Tags { get; } = [];
+
+    // Write this instead:
+    public IReadOnlyList<OrderTag> Tags { get; } = (ImmutableArray<OrderTag>)[];
+
+    // Which is worse than just:
+    public IReadOnlyList<OrderTag> Tags { get; } = Array.Empty<OrderTag>();
+}
+```
+
+First, this immediately fails the simplicity and universality concerns. From our own experience, and the data about the ecosystem, we know that users will need to interface with, well, interfaces :).  This will be an immediate hurdle that will deeply undercut the value of this feature space, potentially even deeply tainting it for the future.
+
+Second, collection expressions would not provide any sort of actual useful syntax or brevity for users in this case.  Because an explicit type would have to be provided, users would be left with syntax with nearly the same complexity and verbosity as what they would have to write today.  Users seeing this would rightfully ask "why would i pick this new form, that is really just the same as the old form, just with some tokens tweaked?"
```

Comment:
```suggestion
Second, collection expressions would not provide any sort of actual useful syntax or brevity for users in this case.  Because an explicit type would have to be provided, users would be left with syntax with nearly the same complexity and verbosity as what they would have to write today.  Users seeing this would rightfully ask "why would I pick this new form, that is really just the same as the old form, just with some tokens tweaked?"
```